services/context-box/server/checksum/checksum_test.go

```diff
@@ -0,0 +1,212 @@
+nodePools:
+  # dynamic nodepools are created by claudie
+  dynamic:
+    # control nodes
+    - name: control-gcp
+      providerSpec:
+        name: gcp0
+        region: europe-west6
+        zone: europe-west6-a
+      count: 2
+      server_type: e2-medium
+      image: ubuntu-os-cloud/ubuntu-2204-jammy-v20221206
+      disk_size: 50
+    - name: compute-gcp
+      providerSpec:
+        name: gcp0
+        region: europe-west6
+        zone: europe-west6-a
+      count: 2
+      server_type: e2-small
+      image: ubuntu-os-cloud/ubuntu-2204-jammy-v20221206
+      disk_size: 50
+    - name: control-oci
+      providerSpec:
+        name: oci-1
+        region: eu-frankfurt-1
+        zone: hsVQ:EU-FRANKFURT-1-AD-1
+      count: 2
+      server_type: VM.Standard2.1
+      # ubuntu minimal
+      # https://docs.oracle.com/en-us/iaas/images/image/674d0b41-aee8-4c0b-bf99-9e100d90f241/
+      image: ocid1.image.oc1.eu-frankfurt-1.aaaaaaaavvsjwcjstxt4sb25na65yx6i34bzdy5oess3pkgwyfa4hxmzpqeq
+      disk_size: 50
+    - name: compute-oci
+      providerSpec:
+        name: oci-1
+        region: eu-frankfurt-1
+        zone: hsVQ:EU-FRANKFURT-1-AD-1
+      count: 2
+      server_type: VM.Standard2.1
+      # ubuntu minimal
+      # https://docs.oracle.com/en-us/iaas/images/image/674d0b41-aee8-4c0b-bf99-9e100d90f241/
+      image: ocid1.image.oc1.eu-frankfurt-1.aaaaaaaavvsjwcjstxt4sb25na65yx6i34bzdy5oess3pkgwyfa4hxmzpqeq
+      disk_size: 50
+
+# kubernetes field is used to define the k8s clusters
+# here we define two clusters, dev and prod
+kubernetes:
+  clusters:
+    - name: og-cluster
+      version: v1.22.0
+      network: 192.168.2.0/24
+      # we can reuse same nodepool spec and claudie will create new nodes
+      pools:
+        control:
+          - control-gcp
+          - control-oci
+        compute:
+          - compute-gcp
+          - compute-oci
+`,
+			out: "325f04bb5560e14e0b162ea3a70aa3368182fff4d433622f3aba20a7837803a8",
+		},
+		{
+			desc: "trololol test",
+			in:   " trololol 
",
+			out:  "754bf6189867b54a764c37acc9cfa8507c43f469f28b28163f63c30487295a68",
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			want, err := hex.DecodeString(tC.out)
+			if err != nil {
+				t.Errorf("Unexpected error: %q", err)
+			}
+
+			if got := CalculateChecksum(tC.in); !bytes.Equal(got, want) {
+				t.Errorf("Unexpected output for %q: expected %+x, actual %+x",
+					tC.desc, want, got)
+			}
+		})
+	}
+}
+
+// TestCompareChecksums tests whether the output of CompareCheckums is what we
+// expect.
+func TestCompareChecksums(t *testing.T) {
+	testCases := []struct {
+		desc string
+		in1  []byte
+		in2  []byte
+		out  bool
+	}{
+		{
+			desc: "false - example short input checksum",
+			in1:  []byte("0123"),
+			in2:  []byte("987543987654543786543029835908643"),
+			out:  false,
+		},
+		{
+			desc: "false - trololol test",
+			in1:  []byte(" trololol 
"),
```

Comment:
Is the `^M` there on purpose?

Also, could we maybe change the `trolol...` to something more formal?

Reply:
> ...is it there on purpose...

yes it is, the other character is ^W, I see it's not rendered here.
I just wanted some non-printable characters there.

> ...could we change...

sure.